src/key.js

```diff
@@ -623,6 +623,86 @@ Key.prototype.revoke = function() {
 };
 
 /**
+ * Signs the public key
+ * @param  {Array<module:key~Key>} privateKey decrypted private keys for signing
+ * @return {module:key~Key} new public key with new certificate signature
+ */
+Key.prototype.sign = function(privateKeys) {
+  var primaryUser, primaryKey, dataToSign, signingKeyPacket, signaturePacket, user, key;
+  if (this.isPrivate()) {
+    throw new Error('Only public key can be signed');
+  }
+  primaryUser = this.getPrimaryUser();
+  if (!primaryUser) {
+    throw new Error('Could not find primary user');
+  }
+  primaryKey = this.primaryKey;
+  dataToSign = {};
+  dataToSign.userid = primaryUser.user.userId || primaryUser.user.userAttribute;
+  dataToSign.key = primaryKey;
```

Comment:
I don't fully understand. So you only sign user IDs and not the public key itself? For example see my uploaded public key:

https://keyserver.ubuntu.com/pks/lookup?op=vindex&search=0xBFB264F6

This brings up the next question... shouldn't we sign all of the key's user IDs as it can have more than one?